### PR TITLE
Form links for modules

### PR DESCRIPTION
**Pull request by orangejenny**
## Summary
https://dimagi-dev.atlassian.net/browse/USH-1197

I didn't rename anything, partly because I couldn't come up with anything less awkward than `FormOrModuleLink`, `FORM_OR_MODULE_WORKFLOW`, etc. and partially because renaming `FormLink` would require running a migration over all apps and builds. Open to disagreement. Maybe `ConditionalLink` would work, since the branching ability is now the main distinction from the other EOF nav workflows.

## Feature Flag
Form linking workflow available on forms

## Product Description
This adds menus to the options available when adding form links:
Screen Shot 2021-09-03 at 6 17 04 PM

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Suite generation has test coverage, UI does not.

### QA Plan
[QA-3499](https://dimagi-dev.atlassian.net/browse/QA-3499)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/helpers/validators.py [hunk 1]
```diff
@@ -767,7 +767,11 @@ def validate_for_build(self, validate_module):
                         self.form.get_app().get_form(form_link.form_id)
                     except FormNotFoundException:
                         errors.append(dict(type='bad form link', **meta))
-            # TODO: add error for module not found
+                else:
+                    try:
+                        self.form.get_app().get_module_by_unique_id(form_link.module_unique_id)
+                    except ModuleNotFoundException:
+                        errors.append(dict(type='bad form link', **meta))
```
Comment (esoergel) [inline corehq/apps/app_manager/helpers/validators.py:774]: I guess `bad form link` is never showed to users anyways?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/app_manager/helpers/validators.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/helpers/validators.py:774]: yeah, it's only used in the massive if-else that is `build_errors.html`

File: corehq/apps/app_manager/views/forms.py [hunk 2]
```diff
@@ -860,6 +840,49 @@ def commtrack_programs():
...
+                and module.root_module.case_type == candidate_module.root_module.case_type
+            )
+            linkable_items.append({
+                'unique_id': candidate_module.unique_id,
+                'name': _module_name(candidate_module),
+                'auto_link': is_top_level or is_child_match,
+            })
+        for candidate_form in candidate_module.get_forms():
+            # Forms can be linked automatically if their module is the same case type as this module,
+            # or if they belong to this module's parent module. All other forms can be linked manually.
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/views/forms.py:872]: sure, didn't realize that initialization and usage were so far apart - 60796b723ca

**Review by esoergel — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
@esoergel would you mind glancing at the last few qabug commits?

**Review by esoergel — APPROVED** [review]

**Pull request merged**